pkg/daemon/metrics.go

```diff
@@ -5,11 +5,55 @@ import (
 	"net/http"
 
 	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
 )
 
-// DefaultBindAddress is the port for the metrics listener
-var DefaultBindAddress = ":8797"
+var (
+	// DefaultBindAddress is the port for the metrics listener
+	DefaultBindAddress = ":8797"
+
+	// RebootCounter is a metric incremented when MCD reboots
+	RebootCounter = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "mcd_reboots",
+			Help: "A metric incremented when MCD reboots",
+		}, []string{"node"})
+
+	// DrainCounter is a metric incremented when MCD drains
+	DrainCounter = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "mcd_drains",
+			Help: "A metric incremented when MCD drains",
```

Comment:
If this is going to be a gauge, it could increment the counter at drain start and decrement at drain finish. Otherwise, should this be a counter like above?

Or also, from my experience looking at that bug yesterday - it would be great to have a metric on how long the drain attempt took. Like, `mcd_drain{node: <node>, state: <"success"|"failure">, err: <error>, duration: <(sec)>}`

Reply:
>  it would be great to have a metric on how long the drain attempt took

this would indeed be great to have